cmd/juju/model/exportbundle.go

```diff
@@ -0,0 +1,113 @@
+// Copyright 2018 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+package model
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/juju/cmd"
+	"github.com/juju/errors"
+	"github.com/juju/gnuflag"
+
+	"github.com/juju/juju/api/bundle"
+	"github.com/juju/juju/cmd/modelcmd"
+	"github.com/juju/juju/cmd/output"
+)
+
+// NewExportBundleCommand returns a fully constructed export bundle command.
+func NewExportBundleCommand() cmd.Command {
+	cmd := &exportBundleCommand{}
+	cmd.newAPIFunc = func() (ExportBundleAPI, error) {
+		return cmd.getAPI()
+	}
+	return modelcmd.Wrap(cmd)
+}
+
+type exportBundleCommand struct {
+	modelcmd.ModelCommandBase
+	out        cmd.Output
+	newAPIFunc func() (ExportBundleAPI, error)
+	Filename   string
+}
+
+const exportBundleHelpDoc = `
+Exports the current model configuration as bundle.
+
+--filename specifies the file where bundle is written to. Its optional.
+If --filename is not used, the bundle is displayed in stdout.
+
+Examples:
+
+    juju export-bundle
+	juju export-bundle --filename mymodel
+`
+
+// Info implements Command.
+func (c *exportBundleCommand) Info() *cmd.Info {
+	return &cmd.Info{
+		Name:    "export-bundle",
+		Purpose: "Exports the current model configuration as bundle.",
+		Doc:     exportBundleHelpDoc,
+	}
+}
+
+// SetFlags implements Command.
+func (c *exportBundleCommand) SetFlags(f *gnuflag.FlagSet) {
+	c.ModelCommandBase.SetFlags(f)
+	c.out.AddFlags(f, "yaml", output.DefaultFormatters)
+	f.StringVar(&c.Filename, "filename", "", "bundle file")
```

Comment:
Please have a look at other example for flags descriptions in the commands. This is meant to be a phrase, starting with a capital :D

Reply:
I have seen cases with --filename. thgats why i chose this.

Reply:
corrected that